src/DynamoCore/Library/FunctionDescriptor.cs

```diff
@@ -318,49 +318,59 @@ public IEnumerable<Tuple<string, string>> InputParameters
             private set;
         }
 
+        private string category;
         /// <summary>
         ///     The category of this function.
         /// </summary>
         public string Category
         {
             get
             {
+                if (category != null)
+                {
+                    return category;
+                }
                 var categoryBuf = new StringBuilder();
                 categoryBuf.Append(GetRootCategory());
 
                 //if this is not BuiltIn function search NodeCategoryAttribute for it
                 if (ClassName != null)
                 {
-                    //get function assembly
-                    var asm = AppDomain.CurrentDomain.GetAssemblies()
-                        .Where(x => x.GetName().Name == Path.GetFileNameWithoutExtension(Assembly))
-                        .ToArray();
-
-                    if (asm.Any() && asm.First().GetType(ClassName) != null)
+                    try
                     {
-                        //get class type of function
-                        var type = asm.First().GetType(ClassName);
-
-                        //get NodeCategoryAttribute for this function if it was been defined
-                        var nodeCat = type.GetMethods().Where(x => x.Name == FunctionName)
-                            .Select(x => x.GetCustomAttribute(typeof(NodeCategoryAttribute)))
-                            .Where(x => x != null)
-                            .Cast<NodeCategoryAttribute>()
-                            .Select(x => x.ElementCategory)
-                            .FirstOrDefault();
-
-                        //if attribute is found compose node category string with last part from attribute
-                        if (!string.IsNullOrEmpty(nodeCat) && (
-                            nodeCat == LibraryServices.Categories.Constructors
-                            || nodeCat == LibraryServices.Categories.Properties
-                            || nodeCat == LibraryServices.Categories.MemberFunctions))
+                        var asm = AppDomain.CurrentDomain.Load(Path.GetFileNameWithoutExtension(Assembly));
```

Comment:
Keep in mind, we will soon have to refactor all assembly load calls throughout Dynamo (because of ALCs)
Any reason for using Load(name) instead of LoadFrom(path) ?

Reply:
So it either Loads it on the spot, or returns it from memory if already loaded right ?

Reply:
I wonder what AppDomain.CurrentDomain.Load will do if we have multiple ALCs with the same assembly name ? is it just the file name or the fully qualified name ?

Reply:
Might the order of assemblies be changed using the dotnet Load API vs what was in there previously?

Reply:
I believe for this code path, all assemblies we are looking for should already be loaded - so there's no new loads happening. Unless something has gone wrong.

What benefit would `LoadFrom` give us?
It seems to me that `LoadFrom` has a higher chance of actually causing a load - which is not what I intend here.

Reply:
I think it defaults to the DefaultALC